ckan/logic/action/get.py

```diff
@@ -1884,9 +1884,12 @@ def package_search(context, data_dict):
         include_drafts = asbool(data_dict.pop('include_drafts', False))
         data_dict.setdefault('fq', '')
         if not include_private:
-            data_dict['fq'] = '+capacity:public ' + data_dict['fq']
+            data_dict['fq'] = 'capacity:public' if not data_dict['fq'] \
+                else 'capacity:public +' + data_dict['fq']
```

Comment:
Same here. What is the purpuse of `+` removal

Reply:
This is the change that was necessary to fix the issue https://github.com/ckan/ckan/issues/4420. The old line created the query `+capacity:public [whatever was in the data dict]`, for example `+capacity:public group:operations`. The `+` at the start of the query is unnecessary because it's only needed to join parts of the query together, not in front of the whole query. There's also no `+` before the second part of the query, which meant it just got ignored. This is why searching for the number of datasets in a particular group returned the number of datasets in *all* groups instead.

Reply:
(This error got into the code when someone tried to switch around the query to make `capacity:public` the first part of it instead of the second, and made a typo. https://github.com/ckan/ckan/commit/107bc894d05d8f2fc0fbcbeddc4b444c20267139)

Reply:
Plus sign inside query doesn't have any `joining` behavior - it's Solr's operator that declares requirement for the following filter. The `+` at the start of the query IS NOT unnecessary - it filters out completely all results that do not contain given `field:value` but otherwise receive high match score because of other query parts.  By writing `else '...  +' ...` you are introducing bug with duplicated `+` that you have mentioned previously.
If there is no `+` before one of `fq` part, it just becomes rather desired, than required, but it's not ignored. Compare [example without plus sign](https://demo.ckan.org/dataset?q=%2Btitle%3ALondon+title%3AAir) and [with plus sign](https://demo.ckan.org/dataset?q=%2Btitle%3ALondon+%2Btitle%3AAir). Both are returning one dataset at the moment i'm writing this comment. But if you remove `title:Air` part completely ([example](https://demo.ckan.org/dataset?q=%2Btitle%3ALondon)) - you'll get two datasets. If parts without `+` are ignored, you would receive the same result in first link

Reply:
Thanks for the detailed explanation, I can see where I got things wrong here now. I will close this PR and make a new one with a proper fix in it.